app/src/main/java/org/wikipedia/page/pageload/PageLoadViewModel.kt

```diff
@@ -0,0 +1,404 @@
+            }
+        } else {
+            setTab(currentTab)
+            currentTab.backStack.add(PageBackStackItem(request.title, request.entry))
+        }
+    }
+
+    private fun switchToExistingTab(position: Int) {
+        if (position < app.tabList.size - 1) {
+            val tab = app.tabList.removeAt(position)
+            app.tabList.add(tab)
+            setTab(tab)
+        }
+
+        if (app.tabCount > 0) {
+            app.tabList.last().squashBackstack()
+            loadFromBackStack()
+        }
+    }
+
+    private fun trimTabCount() {
+        while (app.tabList.size > Constants.MAX_TABS) {
+            app.tabList.removeAt(0)
+        }
+    }
+
+    private fun selectedTabPosition(title: PageTitle): Int {
+        return app.tabList.firstOrNull { it.backStackPositionTitle != null &&
+                title == it.backStackPositionTitle }?.let { app.tabList.indexOf(it) } ?: -1
+    }
+
+    private suspend fun createPageModel(request: PageLoadRequest, response: Response<PageSummary>): PageViewModel {
+        return PageViewModel().apply {
+            curEntry = request.entry
+            forceNetwork = request.options.isRefresh
+            readingListPage = AppDatabase.instance.readingListPageDao().findPageInAnyList(request.title)
+            val pageSummary = response.body()
+            page = pageSummary?.toPage(request.title)
+            title = page?.title
+
+            title?.let {
+                if (!response.raw().request.url.fragment.isNullOrEmpty()) {
+                    it.fragment = response.raw().request.url.fragment
+                }
+                if (it.description.isNullOrEmpty()) {
+                    WikipediaApp.instance.appSessionEvent.noDescription()
+                }
+                if (!it.isMainPage) {
+                    it.displayText = page?.displayTitle.orEmpty()
+                }
+                it.thumbUrl = pageSummary?.thumbnailUrl
+            }
+        }
+    }
+
+    private fun handleError(throwable: Throwable) {
+        L.e(throwable)
+        _pageLoadUiState.value = PageLoadUiState.Error(throwable)
+    }
+
+    private fun determineLoadType(request: PageLoadRequest): LoadType {
+        return when {
+            request.options.shouldLoadFromBackStack -> LoadType.FromBackStack
+            request.options.tabPosition == PageActivity.TabPosition.NEW_TAB_FOREGROUND -> LoadType.NewForegroundTab
+            request.options.tabPosition == PageActivity.TabPosition.NEW_TAB_BACKGROUND -> LoadType.NewBackgroundTab
+            request.options.tabPosition == PageActivity.TabPosition.EXISTING_TAB -> LoadType.ExistingTab
+            else -> LoadType.CurrentTab
+        }
+    }
+
+    companion object {
+        val Factory: ViewModelProvider.Factory = viewModelFactory {
+            initializer {
+                val app = this[APPLICATION_KEY] as WikipediaApp
+                PageLoadViewModel(app)
+            }
+        }
+    }
```

Comment:
Could you please explain it a bit, and what's the difference if we set a `val app = WikipediaApp.instance` directly in the view model class?